### ISSUE TITLE
Lantern client shouldn't forward non-global traffic to proxies

### ISSUE THREAD

**Issue by oxtoacart** [getlantern/lantern#3472]
For example `localhost`.  Proxy should just proxy that traffic directly to the destination.

Right now, we're seeing a lot of requests for localhost, 127.0.0.1, etc. on our proxy servers.

Here's a go function for checking for non-global addresses:

```go
func (server *Server) checkForNonGlobalDestination(addr string) error {
	host, _, err := net.SplitHostPort(addr)
	if err != nil {
		return fmt.Errorf("Unable to split host and port for %v: %v", addr, err)
	}

	ipAddr, err := net.ResolveIPAddr("ip", host)
	if err != nil || len(ipAddr.IP) == 0 {
		return fmt.Errorf("Unable to resolve destination IP addr %v: %v", host, err)
	}

	if !ipAddr.IP.IsGlobalUnicast() {
		return fmt.Errorf("Not accepting connections to non-global address: %s", addr)
	}

	return nil
}
```

**Comment by oxtoacart** [comment]
@aranhoide Opened per your suggestion

**Comment by fffw** [comment]
There's an existing PR https://github.com/getlantern/lantern/pull/3282 implementing similar function.

**Comment by bobrannon1** [comment]
Detains not sure what u mean?


Sent from my Samsung Galaxy smartphone.-------- Original message --------

**Comment by oxtoacart** [comment]
@fffw That detour PR definitely helps, but we also need to avoid localhost when someone turns on "Proxy All" (which is also the default for mobile).

**Comment by myleshorton** [comment]
To be clear @oxtoacart and @fffw this doesn't happen in default mode on desktop right? I.e. the pac file takes care of sending localhost directly right? Definitely looks like that's the case for "127.x" but is it definitely working for "localhost" as well?

**Comment by oxtoacart** [comment]
Yeah, the PAC file works correctly. For example, the Lantern UI is accessible both with 127.0.0.1 and with localhost.

**Comment by myleshorton** [comment]
Cool thanks @oxtoacart!

**Comment by mytharcher** [comment]
Wanna know when could this issue be fixed? I need to block some domain in `/etc/hosts`, such as `127.0.0.1 xxx.com`. But all traffic was passed to Lantern. I am using Lantern 2.1.2 (20160413.044557).

**Comment by myleshorton** [comment]
Thanks @mytharcher I just added this to the backlog for our next sprint, as it's precisely the type of cleanup we'll be focusing on. So hopefully we should have a fix within a month if not sooner.

**Comment by mytharcher** [comment]
@myleshorton really thanks for your awesome work on this project!

**Comment by myleshorton** [comment]
Thanks so much for your kind words @mytharcher

**Comment by myleshorton** [comment]
Related to #4104
